src/wasm/activity-service/activity/activity.go

```diff
@@ -59,14 +64,16 @@ func (a *Activity) ToFIT(options *mesgdef.Options) proto.FIT {
 	fit.Messages = append(fit.Messages, a.Creator.FileId.ToMesg(options))
 	fit.Messages = append(fit.Messages, a.UnrelatedMessages...)
 
-	var totalTimerTime uint32
-	var lastTimestamp time.Time
+	for i := range a.SplitSummaries {
+		mesg := a.SplitSummaries[i].ToMesg(nil)
+		mesg.Fields = append([]proto.Field{
+			factory.CreateField(mesgnum.Session, fieldnum.SessionTimestamp).WithValue(datetime.ToUint32(a.Activity.Timestamp)),
+		}, mesg.Fields...)
```

Comment:
We must add timestamp here, otherwise Garmin Connect rejects it as well. This is a bit weird since split_summary does not actually have defined timestamp, but FIT files produced by Garmin Device contains timestamp. But we don't have control over this, just follow their pattern.